Firestore/Source/API/FIRFirestore.mm

```diff
@@ -243,6 +236,34 @@ - (void)ensureClientConfigured {
       FSTAssert(_settings.host, @"FirestoreSettings.host cannot be nil.");
       FSTAssert(_settings.dispatchQueue, @"FirestoreSettings.dispatchQueue cannot be nil.");
 
+      if (!_settings.enableTimestampsInSnapshots) {
+        FSTWarn(
+            @"The default behavior for NSDate objects stored in Firestore is going to change"
+             "AND YOUR APP MAY BREAK.\n"
+             "To hide this warning and ensure your app does not break, you need to add"
+             "the following code to your app before calling any other Cloud Firestore methods:"
+             "\n\n"
+             "let db = Firestore.firestore()\n"
+             "let settings = db.settings\n"
+             "settings.enableTimestampsInSnapshots = true\n"
+             "db.settings = settings"
+             "\n\n"
+             "With this change, timestamps stored in Cloud Firestore will be read back as"
+             "Firebase Timestamp objects instead of as system Date objects. So you will"
+             "also need to update code expecting a Date to instead expect a Timestamp."
+             "For example:"
+             "\n\n"
+             "// old:\n"
+             "let date: Date = documentSnapshot.data[\"created_at\"]\n"
+             "// new:\n"
+             "let timestamp: Timestamp = documentSnapshot.data[\"created_at\"]\n"
+             "let date: Date = timestamp.approximateDateValue()"
+             "\n\n"
+             "Please audit all existing usages of Date when you enable the new behavior. This new"
+             "behavior will become the default in a future release and so if you do not follow "
```

Comment:
trailing space

Reply:
Unless Objective C concatenates string literals differently from C, it was actually the other way round - other lines were missing spaces, so that the last word from the previous line and the first word of the current line will be lumped together. Fixed.

Reply:
Oh, right! Good call.